bftengine/tests/simpleTest/test_comm_config.cpp

```diff
@@ -0,0 +1,175 @@
+//////////////////////////////////////////////////////////////////////////////
+// Create a replica config for the replica with index `replicaId`.
+// inputReplicaKeyfile is used to read the keys for this replica, and default
+// values are loaded for non-cryptographic configuration parameters.
+void TestCommConfig::GetReplicaConfig(uint16_t replica_id,
+                                      bftEngine::ReplicaConfig* out_config) {
+
+  std::string key_file_name = "private_replica_" + std::to_string(replica_id);
+  std::ifstream keyfile(key_file_name);
+  if (!keyfile.is_open()) {
+    throw std::runtime_error("Unable to read replica keyfile.");
+  }
+
+  bool success = inputReplicaKeyfile(keyfile, key_file_name, *out_config);
+  if (!success)
+    throw std::runtime_error("Unable to parse replica keyfile.");
+
+  // Set non-cryptographic configuration
+  out_config -> statusReportTimerMillisec = 2000;
+  out_config -> concurrencyLevel = 1;
+}
+
+std::unordered_map <NodeNum, NodeInfo> TestCommConfig::SetUpConfiguredNodes(
+    bool is_replica, const std::string& config_file_name, uint16_t id,
+    std::string& ip, uint16_t& port, uint16_t& num_of_clients,
+    uint16_t& num_of_replicas) {
+
+  ConfigFileParser config_file_parser(logger_, config_file_name);
+  if (!config_file_parser.Parse()) {
+    LOG_FATAL(logger_, "Failed to parse configuration file: " <<
+                       config_file_name);
+    exit(-1);
+  }
+  num_of_clients  = static_cast<uint16_t>(
+      config_file_parser.Count(CLIENTS_CONFIG));
+  num_of_replicas = static_cast<uint16_t>(
+      config_file_parser.Count(REPLICAS_CONFIG));
+  if ((is_replica && (id + 1 > num_of_replicas)) ||
+      (!is_replica && (id + 1 > num_of_clients + num_of_replicas))) {
+    LOG_FATAL(logger_, "Wrong number of clients/replicas configured: " <<
+                       "numOfClients=" << num_of_clients <<
+                       ", numOfReplicas=" << num_of_replicas);
+    exit(-1);
+  }
+  vector<string> replicas = config_file_parser.GetValues(REPLICAS_CONFIG);
+  vector<string> clients  = config_file_parser.GetValues(CLIENTS_CONFIG);
+  std::unordered_map <NodeNum, NodeInfo> nodes;
+  int k = 0;
+  for (int i = 0; i < (num_of_replicas + num_of_clients); i++) {
+    vector<string>& current_vector = replicas;
+    if ((k == i) && (i >= num_of_replicas)) { // Ensure this happens only once.
+      current_vector = clients;
+      k = 0;
+    }
+    vector<string> ip_port_pair =
+        config_file_parser.SplitValue(current_vector[k++], ip_port_delimiter_);
+    LOG_INFO(logger_, "setUpConfiguredNodes() id: " << id <<
+                      ", k: " << k - 1 <<
+                      ", port:" << (uint16_t)(std::stoi(ip_port_pair[1])));
+    if (ip_port_pair.size() != 2) {
+      LOG_FATAL(logger_, "Wrong number of parameters configured for "
+                          "replica/client ip/port pair: " <<
+                          ip_port_pair.size());
+      exit(-1);
+    }
+    if (i == id) {
```

Comment:
I might recommend giving `id` a more descriptive name, like `local_id`, or something else to indicate that it's the index of the principal running this code. Not a huge deal, but I found it took me a bit to figure out what its purpose was.

…oh, this is a holdover from the old config functions. Not necessary to fix for this review, but might be a helpful cleanup in the future.

Reply:
Done.